jbi/actions/default.py

```diff
@@ -93,20 +94,20 @@ def __init__(self, steps: dict[Operation, list[Callable]], **parameters):
 
     def __call__(self, context: ActionContext) -> ActionResult:
         """Called from `runner` when the action is used."""
-
-        responses = tuple()  # type: ignore
-
         for step in self.steps[context.operation]:
-            context, step_responses = step(context=context, **self.parameters)
-            for response in step_responses:
-                logger.debug(
-                    "Received %s",
-                    response,
-                    extra={
-                        "response": response,
-                        **context.dict(),
-                    },
-                )
-            responses += step_responses
-
-        return True, {"responses": responses}
+            try:
+                context = context.update(current_step=step.__name__)
+                context = step(context=context, **self.parameters)
+            except IncompleteStepError as exc:
+                context = exc.context
+
+        for response in context.responses:
```

Comment:
I think we'll still want to log after we complete the step and get the responses since in this case, we'd log the same ("complete") context object for each response instead of logging the context as we go.